auth_backends/adfs/opas.py

```diff
@@ -0,0 +1,129 @@
+import base64
+import logging
+import json
+from datetime import datetime
+
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from django.core.cache import cache
+from django.utils.functional import cached_property
+from django.utils.translation import ugettext_lazy as _
+from onelogin.saml2.idp_metadata_parser import OneLogin_Saml2_IdPMetadataParser
+from social_core.backends.saml import SAMLAuth, SAMLIdentityProvider
+
+from tunnistamo.exceptions import FriendlySocialAuthException
+
+
+logger = logging.getLogger(__name__)
+
+
+class NoAssociatedOID(FriendlySocialAuthException):
+    default_message = _('Your AD account does not have an associated OID')
+
+
+class OpasADFS(SAMLAuth):
+    name = 'opas_adfs'
+    metadata_url = 'https://sts.edu.turku.fi/federationmetadata/2007-06/federationmetadata.xml'
+    EXTRA_DATA = ['school_role']
+
+    def generate_saml_config(self, idp=None):
+        ret = super().generate_saml_config(idp)
+        ret['security']['wantAssertionsSigned'] = True,
```

Comment:
Do we want this to be a tuple or is the trailing comma a typo?

Reply:
This comma I took straight from an example, after looking what's in `super().generate_saml_config()` I think it does not need to be there. I tried removing it and generating the config still works, so looks like the example I used has a typo in there.